### DISCUSSION TITLE
Custom server performs differently - next start vs node command

### DISCUSSION THREAD

**Discussion by chandan-reddy-k** [vercel/next.js#16461] [Help]
I have a custom server in my app with the following config (server.js):
```
const express = require('express');
const next = require('next');
const cacheableResponse = require('cacheable-response')
const dev = process.env.NODE_ENV !== 'production';
const port = process.env.PORT || 3000;
const app = next({ dev });
const handle = app.getRequestHandler();

const ssrCache = cacheableResponse({
  ttl: 1000 * 60 * 60, // 1hour
  get: async ({ req, res }) => {
    const data = await app.render(req, res, req.path, {
      ...req.query,
      ...req.params,
    })
    if (res.statusCode === 404) {
      res.end(data)
      return
    }
    return { data }
  },
  send: ({ data, res }) => res.send(data),
})

app.prepare()
  .then(() => {
    const server = express();

    server.get('/', (req, res) => ssrCache({ req, res, pagePath: '/' }))

    server.get('*', (req, res) => {
      return handle(req, res);
    });

    server.listen(port, (err) => {
      if (err) throw err;
      console.log("> Ready on http://localhost:${port}");
    });
  })
  .catch((ex) => {
    console.error(ex.stack);
    process.exit(1);
  });
```

For package.json, I'm getting two different outputs (npm run build && npm run start):

When I use: 
```
"scripts": {
  "dev": "next",
  "build": "next build",
  "start": "next start"
}

```

<img width="1440" alt="Screenshot 2020-08-22 at 8 12 50 AM" src="https://user-images.githubusercontent.com/43699825/90947149-2c9a1200-e451-11ea-848a-39d6add49be2.png">

When I use:
```
"scripts": {
  "dev": "node server.js",
  "build": "next build",
  "start": "NODE_ENV=production node server.js"
}

```

<img width="1438" alt="Screenshot 2020-08-22 at 8 09 45 AM" src="https://user-images.githubusercontent.com/43699825/90947151-302d9900-e451-11ea-981f-e8edb9df3866.png">

As you can see in the images, when I use node server.js command, the _app.js, main.js and index.js are >1MB and the total page size goes >30MB but when I used the next start command, the overall page size ~ 4MB.

Am I forgetting something here and has anyone experienced similar kind of issue?

**Comment by MilesMorel** [comment]
Running into the same thing. I think "start": "NODE_ENV=production node server.js" isn't serving the build at all but just running with the updated env var. My issue was that I want to use esm in my start script....
